toolkit/tools/internal/file/file.go

```diff
@@ -89,6 +89,77 @@ func Copy(src, dst string) (err error) {
 	return NewFileCopyBuilder(src, dst).Run()
 }
 
+// CopyDir copies src directory to dst, creating the dst directory if needed.
+// dst is assumed to be a directory and not a file.
+func CopyDir(src, dst string, newDirPermissions, mergedDirPermissions, childFilePermissions *fs.FileMode) (err error) {
+	isDstExist, err := PathExists(dst)
+	if err != nil {
+		return err
+	}
+	if isDstExist {
+		isDstDir, err := IsDir(dst)
+		if err != nil {
+			return err
+		}
+		if isDstDir {
+			logger.Log.Infof("Destination (%s) already exists and is a directory", dst)
+			if mergedDirPermissions != nil {
+				if err := os.Chmod(dst, *mergedDirPermissions); err != nil {
+					return fmt.Errorf("error setting file permissions: %w", err)
+				}
+			}
+		}
+	}
+
+	if !isDstExist {
+		logger.Log.Infof("Creating destination directory on chroot (%s)", dst)
+		// Create dst dir
+		if newDirPermissions != nil {
+			err = os.MkdirAll(dst, *newDirPermissions)
```

Comment:
Considering that the default permissions is a value `0755`, instead of some other behavior, you could drop the pointer on `newDirPermissions` and `childFilePermissions` and let the caller (i.e. `copyAdditionalDirs()`) handle the default behavior.

Reply:
Is there an added benefit for doing this? I prefer the current implementation since it is clearer.

Reply:
When you say drop the pointer, do you mean to change the signature of the `CopyDir()` function to be this:

`func CopyDir(src, dst string, newDirPermissions, childFilePermissions fs.FileMode, mergedDirPermissions*fs.FileMode) (err error)`

Reply:
Yes.

Reply:
The benefit would be that the default would be stored within the MIC specific code. So, it less likely to be accidentally changed by non-MIC toolkit changes.

Reply:
Added this change in and verified that the functionality described in the description still works.